src/lib/storage/storage_manager.rb

```diff
@@ -58,6 +59,26 @@ def instance
           @instance
         end
 
+        # Use this as an alternative to the instance method.
+        # Probing is skipped and the device tree is initialized from yaml_file.
+        # Any existing probed device tree is replaced.
+        #
+        # @return [::Storage::Storage] libstorage object
+        #
+        def fake_from_yaml(yaml_file = nil)
+          create_instance(
```

Comment:
This call to `create_instance` should be in a separated public method IMHO. See below for cases in which is needed.